utils.go

```diff
@@ -16,3 +18,23 @@ func TransformEach[T any](items []T, fn func(T) Node) []Node {
 	}
 	return nodes
 }
+
+// EscapeCommentContents escapes the contents of a comment node to ensure safe rendering according to https://html.spec.whatwg.org/multipage/syntax.html#comments
+func EscapeCommentContents(s string) string {
```

Comment:
Covers all the major disallowed sequences, but stray “--” inside comments isn’t caught yet per the HTML spec.